lib/MetaCPAN/Script/Snapshot.pm

```diff
@@ -0,0 +1,289 @@
+);
+
+has snap => (
+    is            => 'ro',
+    isa           => Bool,
+    documentation => 'Perform a snapshot',
+);
+
+has list => (
+    is            => 'ro',
+    isa           => Bool,
+    documentation => 'List saved snapshots',
+);
+
+has restore => (
+    is            => 'ro',
+    isa           => Bool,
+    documentation => 'Perform a restore',
+);
+
+## Options
+has snap_stub => (
+    is  => 'ro',
+    isa => 'Str',
+    documentation =>
+        'Stub of snapshot name ( e.g full, user etc ), used with dateformat to create the actual name in S3',
+);
+
+has date_format => (
+    is            => 'ro',
+    isa           => 'Str',
+    documentation => 'strftime format to add to snapshot name (eg %Y-%m-%d)',
+);
+
+has snap_name => (
+    is            => 'ro',
+    isa           => 'Str',
+    documentation => 'Full name of snapshot to restore',
+);
+
+has host => (
+    is            => 'ro',
+    isa           => 'Str',
+    default       => 'http://localhost:9200',
+    documentation => 'ES host, defaults to: http://localhost:9200',
+);
+
+# Note: can take wild cards https://www.elastic.co/guide/en/elasticsearch/reference/2.4/multi-index.html
+has indices => (
+    is      => 'ro',
+    isa     => 'ArrayRef',
+    default => sub { ['*'] },
+    documentation =>
+        'Which indices to snapshot, defaults to "*" (all), can take wild cards - "*v100*"',
+);
+
+## Internal attributes
+
+has aws_key => (
+    is      => 'ro',
+    traits  => ['NoGetopt'],
+    lazy    => 1,
+    default => sub { $_[0]->config->{es_aws_s3_access_key} },
+);
+
+has aws_secret => (
+    is      => 'ro',
+    lazy    => 1,
+    traits  => ['NoGetopt'],
+    default => sub { $_[0]->config->{es_aws_s3_secret} },
+);
+
+has http_client => (
+    is      => 'ro',
+    lazy    => 1,
+    builder => '_build_http_client',
+    traits  => ['NoGetopt'],
+);
+
+sub _build_http_client {
+    return HTTP::Tiny->new(
+        default_headers => { 'Accept' => 'application/json' }, );
+}
+
+## Method selector
+
+sub run {
+    my $self = shift;
+
+    die "es_aws_s3_access_key not in config" unless $self->aws_key;
+    die "es_aws_s3_secret not in config"     unless $self->aws_secret;
+
+    return $self->run_list_snaps if $self->list;
+    return $self->run_setup      if $self->setup;
+    return $self->run_snapshot   if $self->snap;
+    return $self->run_restore    if $self->restore;
+
+    die "setup, restore or snap argument required";
+}
+
+sub run_snapshot {
+    my $self = shift;
+
+    $self->snap_stub   || die 'Missing snap-stub';
+    $self->date_format || die 'Missing date-format (e.g. %Y-%m-%d)';
+
+    my $date      = DateTime->now->strftime( $self->date_format );
+    my $snap_name = $self->snap_stub . '_' . $date;
+
+    my $indices = join ',', @{ $self->indices };
+    my $data = {
+        "indices"              => $indices,
+        "ignore_unavailable"   => 0,
+        "include_global_state" => 1
```

Comment:
Keys don't need to be quoted.  Missing trailing comma.  Could we alpha sort keys as well?